### DISCUSSION TITLE
TeslaFI polling interval

### DISCUSSION THREAD

**Discussion by snooty83** [jhansche/ha-teslafi#34] [Q&A] [answered]
Hello

A question. How is the scan interval set?
So how often is the data retrieved from TeslaFi.com?
Can this be changed somewhere?

Best regards Dani

**Comment by jhansche** [comment, marked as answer]
There is no single answer to this, but there are generally 3 intervals it could be in:

1. While the car state is `sleeping`: 10 minutes
2. While the car state is `driving`: 1 minute
3. Otherwise the default polling interval (when idle and not sleeping) is set to 3 minutes.

Some actions will also poll soon after the commands are sent, in order to wait for the action to settle - for example, some Climate commands, locking/unlocking, and Sentry, will poll again until TeslaFi state reflects the requested state. Those polling delays are:

* If the car was asleep and had to be woken up: poll again in 30 seconds
* For locks and Sentry commands: poll again in 15 seconds
* For Climate commands: poll again in 30 seconds

There is no mechanism currently for adjusting those intervals, and you must keep in mind that TeslaFi has API rate limits.

If you do want to refresh the data more frequently (or ad-hoc, which I do in some cases, both with a script/button in HA as well as in response to certain automation triggers, like opening my garage door), you can trigger the refresh to occur on-demand, using the [Update Entity service](https://www.home-assistant.io/integrations/homeassistant/#service-homeassistantupdate_entity). This service will trigger the `DataUpdateCoordinator` to refresh its data immediately, and as a result _all_ entities in the integration will be updated at that time as well. In other words, you only need to provide one entity to the service, in order to update all entities for the same Vehicle.
